### PR TITLE
feat(profiling): port fzf for search

### PR DESCRIPTION
**Pull request by JonasBa**
Fuse has been struggling with search and on long profiles we often start blocking the main thread... Part of that seems to be because it tries to find all occurrences, so likely runs in O(m*n) time where m is string length and n is query length. We do not actually want it to match that many cases as we've already seen that in some cases the matching was too lax. Fzf seems to be a good substitute as it only ever finds the first occurrence meaning it runs in O(n) time and is pretty fast. On top of that, we also get fine control over search logic (for better or worse) so we could always limit searching via raf with a time budget... For now, this works well, I tested on the profiles that @eliashussary pointed out and we do not seem to be dropping frames anymore

### CODE REVIEW

File: static/app/utils/profiling/fzf/fzf.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+  const textLength = text.length;
+  const patternLength = pattern.length;
+
+  for (let index = 0; index < textLength; index++) {
+    let char = text[index];
+    // This is considerably faster than blindly applying strings.ToLower to the whole string
+    if (!caseSensitive) {
+      const cc = char.charCodeAt(0);
+      if (cc >= 65 && cc <= 90) {
```
Comment (eliashussary) [inline static/app/utils/profiling/fzf/fzf.ts:71]: what are these char codes?

**Review by eliashussary — APPROVED** [review]
i was able to test this out, the results are good and its pretty snappy

taking on our own implementation of `fzf` seems a bit daunting, is there nothing else out there we can take off the shelf?

File: static/app/utils/profiling/fzf/fzf.ts (hunk 1 above)
Comment (JonasBa) [reply static/app/utils/profiling/fzf/fzf.ts:71]: A and Z -> we convert to uppercase to lowercase here

**Comment by JonasBa** [comment]
> i was able to test this out, the results are good and its pretty snappy
> 
> taking on our own implementation of `fzf` seems a bit daunting, is there nothing else out there we can take off the shelf?

Yes, there is https://github.com/ajitid/fzf-for-js, but I decided not to use it as there is still a decent chance that we will block main thread when searching a lot of frames (not as bad as fuse, but not ideal) and the API does not allow us to yield to the main thread. I implemented that in https://github.com/getsentry/sentry/pull/41758 - an api like fzf.search() would not allow us to do that as we cannot break from the iterator

**Pull request merged**